WebScripts~/src/RedisMessagingClient.ts

```diff
@@ -0,0 +1,208 @@
+import { io, Socket, SocketOptions, ManagerOptions } from "socket.io-client";
+
+type RedisMessagingConfig = {
+  url: string;
+  socketIOOptions: SocketOptions & ManagerOptions;
+  isDebug: boolean;
+};
+
+type WebGLGroupListResponse = {
+  status: number;
+  groupListResponse: GroupListResponse;
+}
+
+type GroupListResponse = {
+  groups: Array<{ id: string; name: string }>;
+};
+
+type WebGLCreateGroupResponse = {
+  status: number;
+  createGroupResponse: CreateGroupResponse;
+}
+
+type CreateGroupResponse = {
+  status: number;
+  message: string;
+};
+
+type WebGLJoinResponse = {
+  status: number;
+  message: string;
+}
+
+type Message = {
+  from: string;
+  to: string;
+  messageContent: string;
+};
+
+type RedisMessagingClientCallbacks = {
+  setJoiningGroupStatus: (isJoinedGroup: string) => void;
+  onLeaving: (reason: string) => void;
+  onUnexpectedLeft: (reason: string) => void;
+  onUserJoined: (userId: string) => void;
+  onUserLeaving: (userId: string) => void;
+  onMessageReceived: (message: Message) => void;
+  stopSocket: () => void;
+};
+
+class RedisMessagingClient {
+  private readonly isDebug: boolean;
+
+  private readonly redisMessagingConfig: RedisMessagingConfig;
+  private socket: Socket | null;
+
+  private readonly callbacks: RedisMessagingClientCallbacks;
+
+  constructor(redisMessagingConfig: RedisMessagingConfig, callbacks: RedisMessagingClientCallbacks) {
+    this.socket = null;
+    this.redisMessagingConfig = redisMessagingConfig;
+    this.isDebug = redisMessagingConfig.isDebug;
+    this.callbacks = callbacks;
+  }
+
+  private getSocket = (handle: () => void) => {
```

Comment:
connectErrorHandle等の方が分かりやすいと思いました。
```suggestion
  private getSocket = (connectErrorHandle: () => void) => {
```

Reply:
指摘通り対応しました。
[e12bc3e](https://github.com/extreal-dev/Extreal.Integration.Messaging.Redis/pull/1/commits/e12bc3ed4afbf80f95baf76756bf1d097373079d)

Reply:
確認しました。